wgengine/tstun/tun.go

```diff
@@ -73,8 +89,19 @@ type TUN struct {
 	// filterFlags control the verbosity of logging packet drops/accepts.
 	filterFlags filter.RunFlags
 
-	// insecure disables all filtering when set. This is useful in tests.
-	insecure bool
+	// PreFilterIn is the inbound filter function that runs before the main filter
```

Comment:
Something to think about for a future change: pre- and post- filters that are generic like this feels a bit weird. It feels like we're hardcoding stages that do specific actions, but using generic names. Is it possible to do this as a []FilterFunc instead, with well-defined semantics for how packets traverse the filter stack? (e.g. does Accept end processing, or fall through? What's the default action? etc.)

Reply:
Yes, definitely. I've considered making all filters "uniform" (i.e. having the `FilterFunc` type), which would work reasonably well, but this complicates `SetFilter`, since it's only ever supposed to set the main filter (which may be at a variable position in the filter stack with that model). I'll mull this over more in the next pass.